src/tools/rustfmt/src/macros.rs

```diff
@@ -226,7 +226,9 @@ fn rewrite_macro_inner(
     }
     // Format well-known macros which cannot be parsed as a valid AST.
     if macro_name == "lazy_static!" && !has_comment {
-        if let success @ Some(..) = format_lazy_static(context, shape, ts.trees().collect()) {
+        if let success @ Some(..) =
+            format_lazy_static(context, shape, ts.trees().cloned().collect())
```

Comment:
idk why not simply `ts.clone()`, is this some way of cleaning `Spacing`? Then this should be separate fn instead of that magic.

Reply:
@calebcartwright is this ok to replace with `ts.clone()`? Other way `Spacing` resets to `Spacing::Alone`.

Reply:
there's no magic here. we've toggled the exact code here a few times over the years as the corresponding rustc_* elements were evolved, but only when necessary for compilation purposes and i wouldn't be surprised in the least if it could be changed independent of the broader changes you are proposing in this pr.

feel free to modify the arg as necessary to align with the rustc changes. so long as rustfmt is still compiling and the tests are passing it's fine with me

Reply:
Ok.